packages/react/src/auto/polaris/PolarisAutoFileInput.tsx

```diff
@@ -0,0 +1,175 @@
+    () => metadata.configuration.validations.find((validation) => validation?.__typename === "GadgetRangeFieldValidation"),
+    [metadata]
+  ) as GadgetRangeFieldValidation | undefined;
+
+  useEffect(() => {
+    if (fieldProps.value && isAutoFileFieldValue(fieldProps.value) && fieldProps.value.url) {
+      setImageThumbnailURL(fieldProps.value.url);
+    }
+  }, [fieldProps.value]);
+
+  const onlyImages = useMemo(() => {
+    const imageValidation = metadata.configuration.validations.find(
+      (validation) => validation?.__typename === "GadgetOnlyImageFileFieldValidation"
+    ) as GadgetOnlyImageFileFieldValidation | undefined;
+    if (!imageValidation) return;
+
+    const acceptedTypes = [...imageFileTypes];
+    if (imageValidation.allowAnimatedImages) {
+      acceptedTypes.push("image/gif");
+    }
+
+    return { acceptedTypes, allowAnimatedImages: imageValidation.allowAnimatedImages };
+  }, [metadata]);
+
+  const handleDropZoneDrop = useCallback(
+    async (_dropFiles: File[], acceptedFiles: File[], _rejectedFiles: File[]) => {
+      const file = acceptedFiles[0];
+      const baseFileValue = {
+        fileName: file.name,
+        mimeType: file.type,
+      } satisfies AutoFileFieldValue;
+
+      setImageThumbnailURL(window.URL.createObjectURL(file));
+
+      if (
+        rangeValidation &&
+        ((rangeValidation.min && file.size < rangeValidation.min) || (rangeValidation.max && file.size > rangeValidation.max))
+      ) {
+        fieldProps.onChange({
+          ...baseFileValue,
+          $invalidFileSize: file.size,
+        } satisfies AutoFileFieldValue);
+        return;
+      }
+
+      fieldProps.onChange({
+        ...baseFileValue,
+        $uploading: true,
+      } satisfies AutoFileFieldValue);
+
+      // Upload the file in the background
+      const { url, token } = await (api as any).getDirectUploadToken();
+      await fetch(url, {
+        method: "PUT",
+        headers: {
+          "Content-Type": file.type,
+        },
+        body: file,
+      });
+
+      fieldProps.onChange({
+        directUploadToken: token,
+        ...baseFileValue,
+      } satisfies AutoFileFieldValue);
+    },
+    [api, fieldProps, rangeValidation]
+  );
+
+  const fileUpload = useMemo(() => {
+    if (fieldProps.value) return null;
+
+    const actionHintParts = ["Accepts"];
+    if (onlyImages) {
+      actionHintParts.push(`.jpg, .webp, .svg${onlyImages.allowAnimatedImages ? ", .gif" : ""}, and .png`);
+    }
+
+    if (rangeValidation) {
+      if (rangeValidation.min) {
+        if (rangeValidation.max) {
+          actionHintParts.push(`between ${filesize(rangeValidation.min)} and ${filesize(rangeValidation.max)}`);
+        } else {
+          actionHintParts.push(`larger than than ${filesize(rangeValidation.min)}`);
+        }
+      } else {
+        if (rangeValidation.max) {
+          actionHintParts.push(`smaller than ${filesize(rangeValidation.max)}`);
+        }
+      }
```

Comment:
Seems to overlap with similar validation in `packages/react/src/validationSchema.tsx`